packages/amplify-cli/src/lib/pull-backend.js

```diff
@@ -0,0 +1,38 @@
+const { initializeEnv } = require('./initialize-env');
+const { constructInputParams, postPullCodeGenCheck } = require('./amplify-service-helper');
+
+async function pullBackend(context) {
+  context.exeInfo = context.amplify.getProjectDetails();
+  context.exeInfo.inputParams = constructInputParams(context);
+  context.print.info('');
+  context.print.info('Pre pull check:');
```

Comment:
this should be fired as an event and the check should happen in the event handler

Reply:
currently we don't have pre pull event